### DISCUSSION TITLE
Payload instance settings/preferences

### DISCUSSION THREAD

**Discussion by fryght** [payloadcms/payload#2959] [Feature Requests & Ideas]
As we want to deploy our Payload instance through Docker, it would make sense to be able to specify some settings per deployed instance (ie: Preview URL). Baking this into the admin build means the docker container is no longer portable, so that is not really an option.

These settings should be the same for all users of the admin panel, so using user preferences feels out of place.

Other (admin specific) settings/preferences could also be stored in a similar fashion (ie: Database).

I'm considering working on this myself and to send a pull request.

**Comment by DanRibbens** [comment]
Hi @fryght,

I'm intrigued about your need here. Is there a reason you wouldn't want to use environment variables for your instance based settings?

Alternatively, you could make fields for settings in as globals in your configuration to make it easy for admin users to change, likely with some strict access control functions in place. The downside to this will be that those settings require reading from the DB which might not work for certain requirements.

It is important to rule out all reasonable alternatives before extending the Payload config to keep it simple.
